hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -674,6 +671,7 @@ protected void getAllObjectInternal(List<Data> keys, List<Object> resultingKeyVa
      */
     protected void putAllInternal(Map<?, ?> map) {
         int partitionCount = partitionService.getPartitionCount();
+        int initialSize = map.size() / partitionCount;
```

Comment:
This is a rude "educated" guess to prevent constant resizing of the arrays in `MapEntries`.